torch/distributed/_shard/sharded_tensor/partial_tensor.py

```diff
@@ -152,3 +184,61 @@ def reshard(self, resharding_spec: shard_spec.ShardingSpec) -> ShardedTensor:
             sharded_tensor_size,
             process_group=self.process_group,
         )
+
+    def size(self):
+        return self.local_shard.size()
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if func in _PARTIAL_TENSOR_OPS:
+            return _PARTIAL_TENSOR_OPS[func](types, args, kwargs)
+
+        raise RuntimeError(
+            f"torch function '{func.__name__}', with args: {args} and "
+            f"kwargs: {kwargs} not supported for PartialTensor!")
+
+    def transpose(self, dim0, dim1):
```

Comment:
Maybe after you land this change, I can make Partial Tensor a subclass of tensor too.